### DISCUSSION TITLE
Not possible to get the URL parameter in Filament?

### DISCUSSION THREAD

**Discussion by AllTheUserNamesAreGone** [filamentphp/filament#12246] [Help] [answered]
### Package

Panel builder

### Package Version

v3

### How can we help you?

Hi,

I have an url - /records/2/edit

I have a custom theme and need to get the URL parameter - in this case '2' - in the blade file or the Resource file  - somewhere, anywhere.  I have googled it and tried many many combinations but they always return NULL. I am missing something?

I have tried using app('request'), Request::get, request()->parameter etc etc etc

Any help appreciated. Thanks

**Comment by leandrocfe** [comment]
Share the code you are trying to achieve

**Comment by AllTheUserNamesAreGone** [comment]
I don't have much code to share. It's very simple.
URL:
http://localhost:8000//records/2/edit

Blade File in filament. (But I also can't get the url parameter in the Record ResourceController).

<div>
{{ Request()->parameter }}  or I've also tried a {{ app('request')->request->all() }}  - and many many more combos, also all the Laravel ones that I have found on Google - always I get NULL
</div>

**Reply by dissto** [reply]
`$getRecord()` in your blade maybe? 🤔

**Reply by AllTheUserNamesAreGone** [reply]
Nope - but thanks.

**Comment by AllTheUserNamesAreGone** [comment, marked as answer]
This works!

```{{ request('record') }}```

I found in the Resource file there is this function.

```public static function getPages(): array```

In it defines the routes and also where it defines the parameter key as {record} to access it. 

I did think I saw somewhere in Laravel/Filament there was a specific call or variable to get the param(s) like request()->param. Apparently not.
